### PR TITLE
Fix logic on approval for deployment and oc rollout command upgrade

### PR DESCRIPTION
**Pull request by myeung18**
Signed-off-by: Marco Yeung <myeung@redhat.com>

fixes: upgraded oc deprecated commands, made all pipelines consistent with the way getting approval before promote to production environment.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -3,7 +3,7 @@ pipeline {
         node {
             label 'maven'
         }
-    }
+    }
```
Comment (bmozaffa) [inline Jenkinsfile]: Extra space?

File: nodejsalert-ui/Jenkinsfile [hunk 2]
```diff
@@ -174,8 +189,8 @@ def promoteServiceSetup(openShiftHost, openShiftToken, svcName, registry, imageN
     try {
         sh """ 
             oc create dc ${svcName} --image=${registry}/${imageNameSpace}/${svcName}:${tagName} -n ${projName} 2> /dev/null 
-            oc env dc ${svcName} APP_NAME=${svcName} -n ${projName} 2> /dev/null
-            oc deploy ${svcName} --cancel -n ${projName}  2> /dev/null
+            oc set env dc ${svcName} APP_NAME=${svcName} -n ${projName} 2> /dev/null
+            oc roolout cancel dc ${svcName} -n ${projName}  2> /dev/null
```
Comment (bmozaffa) [inline nodejsalert-ui/Jenkinsfile]: Typo

**Review by bmozaffa — CHANGES_REQUESTED** [review]

File: Jenkinsfile (hunk 1 above)
Comment (myeung18) [reply Jenkinsfile]: space removed

File: Jenkinsfile [hunk 3]
```diff
@@ -324,11 +319,11 @@ stage('Smoke Test') {
...
-        oc env dc ${svcName} MYSQL_SERVICE_NAME=mysql -n ${projName} 2> /dev/null
-        oc env dc ${svcName} MYSQL_SERVICE_USERNAME=${mysqlUser} -n ${projName} 2> /dev/null
-        oc env dc ${svcName} MYSQL_SERVICE_PASSWORD=${mysqlPwd} -n ${projName} 2> /dev/null
-        oc env dc ${svcName} JAVA_APP_DIR=/deployments -n ${projName} 2> /dev/null
-        oc deploy ${svcName} --cancel -n ${projName} -n ${projName} 2> /dev/null
+        oc set env dc ${svcName} MYSQL_SERVICE_NAME=mysql -n ${projName} 2> /dev/null
+        oc set env dc ${svcName} MYSQL_SERVICE_USERNAME=${mysqlUser} -n ${projName} 2> /dev/null
+        oc set env dc ${svcName} MYSQL_SERVICE_PASSWORD=${mysqlPwd} -n ${projName} 2> /dev/null
+        oc set env dc ${svcName} JAVA_APP_DIR=/deployments -n ${projName} 2> /dev/null
+        oc rollout cancel dc ${svcName} -n ${projName} -n ${projName} 2> /dev/null
```
Comment (redhatHameed) [inline Jenkinsfile:326]: I am assuming, these changes will work with the old versions, in case if the client has openshift version <3.7.

Comment (myeung18) [reply Jenkinsfile:326]: According to openshift document, user will need at least OCP 3.3 to have this rollout command work, and oc set env should be fine from OCP 3.2

File: nodejsalert-ui/Jenkinsfile (hunk 2 above)
Comment (myeung18) [reply nodejsalert-ui/Jenkinsfile]: fixed and tested.

**Review by bmozaffa — APPROVED** [review]

**Pull request merged**